bootstrap/terraform/main.tf

```diff
@@ -1,189 +1,144 @@
+
 locals {
+  name            = var.name
+  region          = var.region
+  
   cluster_version = var.cluster_version
-  azs             = data.aws_availability_zones.available.names
-
-  vpc_cidr        = var.vpc_cidr
-  vpc_name        = join("-", [var.name, "vpc"])
-  cluster_name    = join("-", [var.name, "eks"])
-  node_group_name = "mng-ondemand"
-}
-
-#---------------------------------------------------------------
-# This vpc module creates VPC, 3 Private Subnets, 3 Public Subnets, IGW, Single NAT gateway
-# You can comment or remove module if you already have an existing VPC and Subnets. You must add public_subnet_tags, private_subnet_tags to your existing VPC
-#---------------------------------------------------------------
-module "vpc" {
-  source  = "terraform-aws-modules/vpc/aws"
-  version = "v3.2.0"
-
-  name = local.vpc_name
-  cidr = local.vpc_cidr
-  azs  = local.azs
-
-  public_subnets  = [for k, v in slice(data.aws_availability_zones.available.names, 0, 2) : cidrsubnet(local.vpc_cidr, 8, k)]
-  private_subnets = [for k, v in slice(data.aws_availability_zones.available.names, 0, 2) : cidrsubnet(local.vpc_cidr, 8, k + 10)]
-
-  enable_nat_gateway   = true
-  create_igw           = true
-  enable_dns_hostnames = true
-  single_nat_gateway   = true
-
-  public_subnet_tags = {
-    "kubernetes.io/role/elb" = "1"
-  }
-
-  private_subnet_tags = {
-    "kubernetes.io/role/internal-elb" = "1"
+  cluster_name    = local.name
+  
+  crossplane_helm_config = var.crossplane_helm_config
+  crossplane_aws_provider = var.crossplane_aws_provider
+  crossplane_kubernetes_provider = var.crossplane_kubernetes_provider
+  
+  vpc_name        = local.name
+  vpc_cidr = "10.0.0.0/16"
+  azs      = slice(data.aws_availability_zones.available.names, 0, 2)
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/awslabs/crossplane-on-eks"
   }
 }
 
 #---------------------------------------------------------------
-# This module deploys EKS Cluster with one Managed group
+# EKS Blueprints
 #---------------------------------------------------------------
+
 module "eks_blueprints" {
-  source = "github.com/aws-ia/terraform-aws-eks-blueprints"
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.18.0"
+
+  # EKS CONTROL PLANE VARIABLES
+  cluster_name    = local.cluster_name
+  cluster_version = local.cluster_version
 
   # EKS Cluster VPC and Subnet mandatory config
   vpc_id             = module.vpc.vpc_id
   private_subnet_ids = module.vpc.private_subnets
 
-  # EKS CONTROL PLANE VARIABLES
-  create_eks      = true
-  cluster_version = local.cluster_version
-  cluster_name    = local.cluster_name
-
   # EKS MANAGED NODE GROUPS
   managed_node_groups = {
-    mg_5 = {
-      node_group_name = local.node_group_name
-      instance_types  = ["m5.xlarge"]
-      min_size        = "1"
+    mg = {
+      node_group_name = "managed-on-demand"
+      instance_types  = ["t3.small"]
+      desired_size    = 2
```

Comment:
I meant this one to be changed to `min_size`. 
desired_size wont have any effect by the Terraform once the node group is created. This field will be taken over by auto scaling groups. We can only modify either min or max size from Terraform